From: Jiri Slaby <jslaby@suse.cz>
Subject: kABI: protect jiffies types
Patch-mainline: never, kabi
References: kabi

In 4.4.127, commit 16d18bf7a173bac1993426e8769389d9eb79b261 (jiffies.h:
declare jiffies and jiffies_64 with ____cacheline_aligned_in_smp),
changed alignment and section placement of jiffies and jiffies_64. As it
does not change the type, no modules should complain about this.

So protect the change by __GENKSYMS__ as usual.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 include/linux/jiffies.h |    6 ++++++
 1 file changed, 6 insertions(+)

--- a/include/linux/jiffies.h
+++ b/include/linux/jiffies.h
@@ -73,8 +73,14 @@ extern int register_refined_jiffies(long
  * without sampling the sequence number in jiffies_lock.
  * get_jiffies_64() will do this for you as appropriate.
  */
+#ifdef __GENKSYMS__
+#define __jiffy_data  __attribute__((section(".data")))
+extern u64 __jiffy_data jiffies_64;
+extern unsigned long volatile __jiffy_data jiffies;
+#else
 extern u64 __cacheline_aligned_in_smp jiffies_64;
 extern unsigned long volatile __cacheline_aligned_in_smp __jiffy_arch_data jiffies;
+#endif
 
 #if (BITS_PER_LONG < 64)
 u64 get_jiffies_64(void);
